ql/math/randomnumbers/burley2020sobolrsg.cpp

```diff
@@ -0,0 +1,153 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Peter Caspers
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/randomnumbers/burley2020sobolrsg.hpp>
+#include <ql/math/randomnumbers/mt19937uniformrng.hpp>
+#include <ql/errors.hpp>
+
+namespace QuantLib {
+
+    Burley2020SobolRsg::Burley2020SobolRsg(Size dimensionality,
+                                           unsigned long seed,
+                                           SobolRsg::DirectionIntegers directionIntegers,
+                                           unsigned long scrambleSeed)
+    : dimensionality_(dimensionality), seed_(seed), directionIntegers_(directionIntegers),
+      integerSequence_(dimensionality), sequence_(std::vector<Real>(dimensionality), 1.0) {
+        reset();
+        group4Seeds_.resize((dimensionality_ - 1) / 4 + 1);
+        MersenneTwisterUniformRng mt(scrambleSeed);
+        for (auto& s : group4Seeds_) {
+            s = static_cast<std::uint32_t>(mt.nextInt32());
+        }
+    }
+
+    void Burley2020SobolRsg::reset() const {
+        sobolRsg_ = ext::make_shared<SobolRsg>(dimensionality_, seed_, directionIntegers_, false);
+        nextSequenceCounter_ = 0;
+    }
+
+    const std::vector<std::uint32_t>& Burley2020SobolRsg::skipTo(std::uint32_t n) const {
+        reset();
+        for (Size k = 0; k < n + 1; ++k) {
```

Comment:
This doesn't actually skip to the required sample — it calculates and discards all previous ones.  Is this method here because there's some common code that uses either Sobol or Burley-Sobol and requires it?

Reply:
True. I don't see a smart way to fast-forward the burley sequence. I still wanted to provide this method so that Burley2020SobolRsg can be used as a one-to-one replacement for SobolRsg.

Reply:
ok